sdk/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Alpha.hs

```diff
@@ -27,6 +32,7 @@ data AlphaEnv = AlphaEnv
   , boundExprVarsRhs :: !(Map.Map ExprVarName Int)
     -- ^ Maps bound expr variables from the right-hand-side to
     -- the depth of the binder which introduced them.
+  , tconEquivalence :: !(Qualified TypeConName -> Qualified TypeConName -> Bool)
```

Comment:
I opted to add this to the environment, because the way the environment is used currently is pretty much identical to the reader monad - I could spin it out for very minor improvement to two lines where we call bindTypeVar, but it would require about 200 lines of code to pass around a new parameter.

Reply:
I've added a comment to make it clear tconEquivalence shouldn't mutate